src/java_bytecode/jar_file.h

```diff
@@ -45,18 +45,26 @@ class jar_filet
 protected:
   mz_zip_archive zip;
   bool mz_ok;
+  std::string matcher;
+  std::map<int, int> index_map;
 };
 
-class jar_poolt
+class jar_poolt:public messaget
 {
 public:
+  void set_java_cp_include_files(std::string &_java_cp_include_files)
+  {
+    java_cp_include_files=_java_cp_include_files;
+  }
   jar_filet &operator()(const std::string &file_name)
   {
+    assert(!java_cp_include_files.empty() && "class regexp cannot be empty");
     file_mapt::iterator it=file_map.find(file_name);
     if(it==file_map.end())
     {
       jar_filet &jar_file=file_map[file_name];
-      jar_file.open(file_name);
+      jar_file.set_message_handler(get_message_handler());
```

Comment:
I'll add a status output to see about hidden files, also the JSON parser needs a valid one